### PR TITLE
Formatting of local links in text; Lists of Figures, Tables and Examples

### PR DESCRIPTION
**Pull request by davidhassell**
See issue #499 for discussion of these changes.

# Release checklist
- [x] Authors updated in `cf-conventions.adoc`? Add in two places: on line 3 and under `.Additional Authors` in `About the authors`.
- [x] Next version in `cf-conventions.adoc` up to date? Versioning inspired by [SemVer](https://semver.org).
- [x] `history.adoc` up to date?
- [x] Conformance document up to date?

# For maintainers
After the merge remember to delete the source branch.
Tags are set at the conclusion of the annual meeting; until then, `main` always is a draft for the next version.

### CODE REVIEW

File: ch03.adoc [hunk 1]
```diff
@@ -139,7 +139,7 @@ Use of these attributes for data packing, which is their most important applicat
 [[long-name, Section 3.2, "Long Name"]]
 === Long Name
 
-The **`long_name`** attribute is defined by the NUG to contain a long descriptive name which may, for example, be used for labeling plots.
+The **`long_name`** attribute is defined by the NUG <<NUG>> to contain a long descriptive name which may, for example, be used for labeling plots.
```
Comment (JonathanGregory) [inline ch03.adoc:142]: could this be just <<NUG>>?

File: ch05.adoc [hunk 2]
```diff
@@ -20,7 +20,7 @@ We recommend that the name of a multidimensional coordinate variable should not
 This practice also avoids potential bugs in applications that determine coordinate variables by only checking for a name match between a dimension and a variable and not checking that the variable is one dimensional.
 
 If the longitude, latitude, vertical or time coordinate is multi-valued, varies in only one dimension, and varies independently of other spatiotemporal coordinates, it is not permitted to store it as an auxiliary coordinate variable.
-This is both to enhance conformance to COARDS and to facilitate the use of generic applications that recognize the NUG convention for coordinate variables.
+This is both to enhance conformance to COARDS and to facilitate the use of generic applications that recognize the NUG <<NUG>> convention for coordinate variables.
```
Comment (JonathanGregory) [inline ch05.adoc:23]: `<<NUG>>`?

File: ch09.adoc [hunk 3]
```diff
@@ -115,13 +114,15 @@ The **orthogonal multidimensional array representation**, the simplest represent
...
 In both examples, there will be longitude and latitude coordinate variables, x(i), y(i), that are one-dimensional and defined along the instance dimension.
 
-Table 9.2 illustrates the storage of a data variable using the orthogonal multidimensional array representation.
+<<table-orthogonal-array, Table 9.2>> illustrates the storage of a data variable using the orthogonal multidimensional array representation.
 The data variable holds a collection of 4 features.
 The individual features, distinguished by color, are sequenced along the horizontal axis by the instance dimension indices, i1, i2, i3, i4.
 Each instance contains three elements, sequenced along the vertical with element dimension indices, o1, o2, o3.
-The i and o subscripts would be interchanged (i.e. Table 9.2 would be transposed) if the element dimension were the netCDF unlimited dimension.
+The i and o subscripts would be interchanged (i.e. <<table-orthogonal-array, Table 9.2>> would be transposed) if the element dimension were the netCDF unlimited dimension.
```
Comment (JonathanGregory) [inline ch09.adoc:121]: I don't think we need "Table 9.2" as well as `<<table-orthogonal-array>>`

File: ch09.adoc [hunk 4]
```diff
@@ -178,15 +177,17 @@ The **incomplete multidimensional array representation** can used if the feature
 That is, features that are shorter than the longest feature must be padded with missing values to bring all instances to the same storage size.
 This representation sacrifices storage space to achieve simplicity for reading and writing.  
 
-Table 9.3 illustrates the storage of a data variable using the orthogonal multidimensional array representation.  
+<<table-incomplete-array, Table 9.3>> illustrates the storage of a data variable using the orthogonal multidimensional array representation.  
 The data variable holds a collection of 4 features.
 The individual features, distinguished by color, are sequenced by the instance dimension indices, i1, i2, i3, i4.
 The instances contain respectively 2, 4, 3 and 6 elements, sequenced by the element dimension index with values of o1, o2, o3, ... .
-The i and o subscripts would be interchanged (i.e. Table 9.3 would be transposed) if the element dimension were the netCDF unlimited dimension.
+The i and o subscripts would be interchanged (i.e. <<table-incomplete-array, Table 9.3>> would be transposed) if the element dimension were the netCDF unlimited dimension.
```
Comment (JonathanGregory) [inline ch09.adoc:184]: I don't think we need "Table 9.3" as well as the symbolic ref.

File: ch09.adoc [hunk 5]
```diff
@@ -232,14 +233,14 @@ The i and o subscripts would be interchanged (i.e. Table 9.3 would be transposed
 {set:cellbgcolor:#ddaaaa}
 |===============
 
-Table 9.3.   The storage of data using the incomplete multidimensional array representation (subscripts in CDL order).
-
 ====  Contiguous ragged array representation
 
 The **contiguous ragged array representation** can be used only if the size of each feature is known at the time that it is created. 
-In this representation the data for each feature will be contiguous on disk, as shown in Table 9.4.
+In this representation the data for each feature will be contiguous on disk, as shown in <<table-contiguous-array, Table 9.4>>.
```
Comment (JonathanGregory) [inline ch09.adoc:239]: I don't think we need "Table 9.4" as well as the symbolic ref.

File: ch09.adoc [hunk 6]
```diff
@@ -316,11 +315,20 @@ For indices that correspond to features, whose data have not yet been written, t
 
 ==== Indexed ragged array representation
 
-The **indexed ragged array representation** stores the features interleaved along the sample dimension in the data variable as shown in Table 9.4.
+The **indexed ragged array representation** stores the features interleaved along the sample dimension in the data variable as shown in <<table-indexed-array, Table 9.5>>.
```
Comment (JonathanGregory) [inline ch09.adoc:318]: I don't think we need "Table 9.5" as well as the symbolic ref.

File: ch08.adoc [hunk 7]
```diff
@@ -17,7 +17,7 @@ The disadvantage is that generic utilities that don't recognize the CF conventio
 === Packed Data
 
 At the current time the netCDF interface does not provide for packing data.
-However a simple packing may be achieved through the use of the optional NUG defined attributes **`scale_factor`** and **`add_offset`**.
+However a simple packing may be achieved through the use of the optional NUG <<NUG>> defined attributes **`scale_factor`** and **`add_offset`**.
```
Comment (JonathanGregory) [inline ch08.adoc:20]: just `<<NUG>>`?

File: ch03.adoc (hunk 1 above)
Comment (davidhassell) [reply ch03.adoc]: Fixed: 3df131ec211a578713034ae3f65518fe202e1a96

**Comment by davidhassell** [comment]
Dear @JonathanGregory,

PR #547 caused some big clashes, but I think I've sorted it. If you could briefly look over section 7.1 before merging, that would be great.

Many thanks,
David

**Comment by JonathanGregory** [comment]
As far as I can tell it's correct - but seeing is believing, so I'll merge it now. Thanks for doing a fiddly job conscientiously.

**Pull request merged**